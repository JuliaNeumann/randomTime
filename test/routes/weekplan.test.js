import { shallow } from 'vue-test-utils';
import Weekplan from '../../app/routes/weekplan/weekplan.vue';

describe('Weekplan', () => {
  const wrapper = shallow(Weekplan);

  it('renders the correct markup', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('triggers request for setting a weekplan on click and displays success message', () => {
    global.axios = {};
    axios.post = () => {
      return {
        then: (callback) => {
          callback();
          return {
            catch: () => {
              return true;
            }
          }
        }
      }
    };

    wrapper.find('input[type="submit"]').trigger('click');
    const successMsg = wrapper.find('.uk-alert-primary');
    expect(successMsg).toBeDefined;

    const errorMsg = wrapper.find('.uk-alert-danger');
    expect(errorMsg).toBeUndefined;
  });

  it('triggers request for setting a weekplan on click and displays error message', () => {
    global.axios = {};
    axios.post = () => {
      return {
        then: () => {
          return {
            catch: (callback) => {
              callback({data: 'error'});
            }
          }
        }
      }
    };

    wrapper.find('input[type="submit"]').trigger('click');
    const successMsg = wrapper.find('.uk-alert-primary');
    expect(successMsg).toBeUndefined;

    const errorMsg = wrapper.find('.uk-alert-danger');
    expect(errorMsg).toBeDefined;
    expect(errorMsg.text()).toBe('error');
  });

});