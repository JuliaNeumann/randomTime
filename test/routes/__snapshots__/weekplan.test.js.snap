// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekplan renders the correct markup 1`] = `
<div
  class="uk-container"
  id="app"
>
  <!---->
   
  <div>
    <h1
      class="uk-heading-divider"
    >
      Create Week Plan
    </h1>
     
    <form
      class="uk-form-stacked"
    >
      <fieldset
        class="uk-fieldset"
      >
        <label
          class="uk-form-label"
          for="number-of-hours"
        >
          Number of hours to plan:
        </label>
         
        <div
          class="uk-form-controls"
        >
          <input
            class="uk-input uk-form-width-medium"
            id="number-of-hours"
            type="number"
          />
        </div>
         
        <label
          class="uk-form-label uk-margin"
          for="user"
        >
          User:
        </label>
         
        <div
          class="uk-form-controls"
        >
          <input
            class="uk-input uk-form-width-medium"
            id="user"
            type="text"
          />
        </div>
         
        <input
          class="uk-button uk-button-primary uk-margin"
          type="submit"
          value="Create"
        />
      </fieldset>
    </form>
     
    <!---->
     
    <!---->
     
    <!---->
  </div>
</div>
`;
